src/protocol/requests/api-versions/api-versions-request-builder.ts

```diff
@@ -0,0 +1,76 @@
+import { TagSection } from '../../commons';
+import { type CompactString, Int16, Int32, type NullableString } from '../../primitives';
+import { type RequestHeader, RequestHeaderV1, RequestHeaderV2 } from '../headers';
+import { type RequestBuilder } from '../request-builder';
+import { ApiVersionsRequestV0 } from './api-versions-request-v0';
+import { ApiVersionsRequestV1 } from './api-versions-request-v1';
+import { ApiVersionsRequestV2 } from './api-versions-request-v2';
+import { ApiVersionsRequestV3 } from './api-versions-request-v3';
+
+export type ApiVersionsRequest =
+  | ApiVersionsRequestV0
+  | ApiVersionsRequestV1
+  | ApiVersionsRequestV2
+  | ApiVersionsRequestV3;
+
+export class ApiVersionsRequestBuilder implements RequestBuilder<ApiVersionsRequest> {
+  private static readonly apiKey = 18;
+
+  constructor(
+    private readonly clientId: NullableString,
+    private readonly clientSoftwareName: CompactString,
+    private readonly clientSoftwareVersion: CompactString
+  ) {}
+
+  public getApiKey(): number {
+    return ApiVersionsRequestBuilder.apiKey;
+  }
+
+  public build(correlationId: number, maxVersion: number): { request: ApiVersionsRequest; header: RequestHeader } {
+    switch (maxVersion) {
```

Comment:
why is this parameter named `maxVersion`, while the behavior suggest it to be the exact version?